### PR TITLE
[FIX] MSSimulator thread-safe now

### PR DESCRIPTION
**Pull request by cbielow**
on rare occasions, MSSimulator would crash when using >1 threads, since Boost RNGs are not threadsafe.
This is fixed now.

Also fixes a `bad_alloc` when the user specifies protein intensities manually which are numeric_limits<uint>. This is now resulting in
```
exception of type InvalidValue occured in line 325, function void __cdecl OpenMS::IonizationSimulation::ionizeEsi_(class OpenMS::FeatureMap &,class OpenMS::ConsensusMap &) of C:\dev\openmsqt5\src\openms\source\SIMULATION\IonizationSimulation.cpp
error message: the value '6.67539e+18' was used but is not valid; Protein abundance is too high. Please use values in [0,4294967295]!
```

### CODE REVIEW

**Comment by timosachsenberg** [comment]
looks good

File: src/openms/source/SIMULATION/IonizationSimulation.cpp [hunk 1]
```diff
@@ -288,179 +287,194 @@ namespace OpenMS
...
+
+            if (charge == 0)
+            {
+              continue;
+            }
+
+            /////
+// distribute charges across adduct types
+/////
+Compomer cmp;
```
Comment (timosachsenberg) [inline src/openms/source/SIMULATION/IonizationSimulation.cpp]: indentation?

**Comment by timosachsenberg** [comment]
please add a test for MSSimulator

File: src/openms/source/SIMULATION/IonizationSimulation.cpp (hunk 1 above)
Comment (cbielow) [reply src/openms/source/SIMULATION/IonizationSimulation.cpp]: fixed

**Pull request merged**